test/src/module/common/ioTlsTest.c

```diff
@@ -0,0 +1,218 @@
+        harnessTlsServerAccept();
+        harnessTlsServerClose();
+
+        // Success on valid ca file and match common name
+        harnessTlsServerAccept();
+        harnessTlsServerClose();
+
+        // Success on valid ca file and match alt name
+        harnessTlsServerAccept();
+        harnessTlsServerClose();
+
+        // Unable to find matching hostname in certificate
+        harnessTlsServerAccept();
+        harnessTlsServerClose();
+
+        // Certificate error
+        harnessTlsServerAccept();
+        harnessTlsServerClose();
+
+        // Certificate ignored
+        harnessTlsServerAccept();
+        harnessTlsServerClose();
+
+        exit(0);
+    }
+}
+
+/***********************************************************************************************************************************
+Test server
+***********************************************************************************************************************************/
+static void
+testTlsServer(void)
+{
+    if (fork() == 0)
+    {
+        harnessTlsServerInit(TLS_TEST_PORT, TLS_CERT_TEST_CERT, TLS_CERT_TEST_KEY);
+
+        // First protocol exchange
+        harnessTlsServerAccept();
+
+        harnessTlsServerExpect("some protocol info");
+        harnessTlsServerReply("something:0\nsome content");
+
+        // This will cause the client to disconnect
+        sleepMSec(500);
+
+        // Second protocol exchange
+        harnessTlsServerExpect("more protocol info");
+        harnessTlsServerReply("0123456789AB");
+
+        harnessTlsServerClose();
+
+        exit(0);
+    }
+}
+
+/***********************************************************************************************************************************
+Test Run
+***********************************************************************************************************************************/
+void
+testRun(void)
+{
+    FUNCTION_HARNESS_VOID();
+
+    // Additional coverage not provided by testing with actual certificates
+    // *****************************************************************************************************************************
+    if (testBegin("asn1ToStr(), tlsClientHostVerify(), and tlsClientHostVerifyName()"))
+    {
+        TEST_ERROR(asn1ToStr(NULL), CryptoError, "TLS certificate name entry is missing");
+
+        TEST_ERROR(
+            tlsClientHostVerifyName(
+                strNew("host"), strNewN("ab\0cd", 5)), CryptoError, "TLS certificate name contains embedded null");
+
+        TEST_ERROR(tlsClientHostVerify(strNew("host"), NULL), CryptoError, "No certificate presented by the TLS server");
+
+        TEST_RESULT_BOOL(tlsClientHostVerifyName(strNew("host"), strNew("**")), false, "invalid pattern");
+        TEST_RESULT_BOOL(tlsClientHostVerifyName(strNew("host"), strNew("*.")), false, "invalid pattern");
+        TEST_RESULT_BOOL(tlsClientHostVerifyName(strNew("a.bogus.host.com"), strNew("*.host.com")), false, "invalid host");
```

Comment:
1) Would like to see a couple tests where tlsClientVerifyName is true: www.host.com, *.host.com should return true.
2) Also, are we supporting Subject Alternative Names such that host.com would match *.host.com (see https://en.wikipedia.org/wiki/Wildcard_certificate and https://en.wikipedia.org/wiki/Subject_Alternative_Name)

Reply:
There are more tests below in run 2.  It doesn't seem worth it to duplicate.  As the comment says, `Additional coverage not provided by testing with actual certificates`.

We are following RFC2818 here -- as Postgres does.  In fact this code is pulled from Postgres.  The wiki seems to say the same thing as I read it: `*.domain.com is OK. It will match www.domain.com but not domain.com and not zzz.www.domain.com`

Reply:
The wiki and RFC also mention partial wildcards but we don't support them.